external-crates/move/crates/move-package/src/resolution/dependency_graph.rs

```diff
@@ -240,6 +245,24 @@ impl<Progress: Write> DependencyGraphBuilder<Progress> {
             })
             .unwrap_or(None);
 
+        // add implicit dependencies to the manifest. Implicit dependencies are added only to the
+        // root package here; at the end of this method (after the graph is built) we iterate over
+        // all packages in the graph and add the dependencies there. The reason we add them to the
+        // root package here is to ensure that they and their dependencies are correctly processed.
+        //
+        // implicits deps should be skipped entirely if the root manifest contains any of them explicitly
+        let add_implicits: bool = self
```

Comment:
Are we missing a negation here? I.e. we should only add implicits if all the implicits are not mentioned in the root?